From 9647a18e741d981dc01c13f11099c15f6204126d Mon Sep 17 00:00:00 2001
From: Jing Hua <jinghua@marvell.com>
Date: Fri, 13 Jun 2014 09:59:28 +0800
Subject: [PATCH 1711/1825] fix: alp: tpm: for HWF rules with vlan number,
 keep pbit if it is required

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit c428cedc9d294b4b0b424eb002dc88587ac6db5b

        for HWF rules with vlan number as key, keep pbit as a match field
        if that is configured by TPM API.
        SYSTEMSW-690 - <TPM cant support Pbit value and vlan num rules checking hardware forward rule>

Signed-off-by: Jing Hua <jinghua@marvell.com>

Change-Id: I306ed35e83b0342795cf9e3d9f39031b4857012d
Reviewed-on: http://vgitil04.il.marvell.com:8080/8521
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Ken Ma <make@marvell.com>
Reviewed-by: Dmitri Epshtein <dima@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../mv_drivers_lsp/mv_tpm/inc/tpm_mng.h            |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_tpm/inc/tpm_mng.h b/arch/arm/plat-armada/mv_drivers_lsp/mv_tpm/inc/tpm_mng.h
index 17abbed..6c36c6e 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_tpm/inc/tpm_mng.h
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_tpm/inc/tpm_mng.h
@@ -83,8 +83,7 @@ disclaimer.
 #define TPM_MNG_PON_QHIGH  3
 
 /* all vlan field_bm */
-#define TPM_MNG_ALL_VLAN_FIELD_BM  (TPM_MATCH_VID_INNER | TPM_MATCH_VID_OUTER \
-				   | TPM_MATCH_PBITS_OUTER | TPM_MATCH_PBITS_INNER)
+#define TPM_MNG_ALL_VLAN_FIELD_BM  (TPM_MATCH_VID_INNER | TPM_MATCH_VID_OUTER)
 
 /* rule_type value for API type other than CLS */
 #define TPM_MNG_RULE_TYPE_INVALID	0
-- 
1.7.5.4

